pkg/installer/generateconfig.go

```diff
@@ -31,6 +32,20 @@ import (
 	"github.com/Azure/ARO-RP/pkg/util/version"
 )
 
+type capabilityChecker func(*mgmtcompute.ResourceSku, string) bool
```

Comment:
nit: I don't think we need this.  The only method we're passing to the supportedDefaultDisk is `computeskus.HasCapability` and I don't see that changing.  

IMO remove the argument from the `supportedDefaultDisk function and just call it directly.  Would make things a bit cleaner, unless you can justify a use case of passing it as an argument.

Reply:
I guess it makes testing easier, but imo decreases readability.  In your test you could just pass in a vmskus object with premium const or without it.

Reply:
Testing was my primary concern when I originally decided to pass the `HasCapability` function in but I was able to do a little more setup in the test itself to avoid having to mock the function out.